File: algorithms/sorting/selection-sort/scala/SelectionSort.scala
object SelectionSort {
  def sort(arr: Array[Int]): Unit = {
    val n = arr.length
    for (i <- 0 until n - 1) {
      var min_idx = i
      for (j <- i + 1 until n) {
        if (arr(j) < arr(min_idx))
          min_idx = j
      }
      val temp = arr(min_idx)
      arr(min_idx) = arr(i)
      arr(i) = temp
    }
  }
}

File: algorithms/sorting/selection-sort/go/selection_sort.go
package selectionsort

// SelectionSort sorts an array of integers using the Selection Sort algorithm.
func SelectionSort(arr []int) {
	n := len(arr)
	for i := 0; i < n-1; i++ {
		min_idx := i
		for j := i + 1; j < n; j++ {
			if arr[j] < arr[min_idx] {
				min_idx = j
			}
		}
		arr[i], arr[min_idx] = arr[min_idx], arr[i]
	}
}
